pynetdicom/tests/test_ae.py

```diff
@@ -477,16 +516,35 @@ def handle(event):
         assert assoc.is_aborted
         assert len(scp.active_associations) == 0
 
-        scu_ae.acse_timeout = 0
+        # FIXME: If this is `0` we can process an ABORT primitive where
```

Comment:
Does this FIXME still need fixing?

Reply:
Yes, this breaks the state machine:

```
ERROR    pynetdicom.sm:fsm.py:100 State Machine received an exception attempting to perform the action 'AE-2' while in state 'Sta4'
ERROR    pynetdicom.sm:fsm.py:103 'A_ABORT' object has no attribute 'calling_ae_title'
```

---

I looked into this a bit - it seems like there are some states where we expect the "current primitive" to be unchanged from one state to the next. E.g: when sending an associate rq as a SCU, we expect the primitive to still be there after we successfully connect. The fix probably boils down to addressing this comment https://github.com/pydicom/pynetdicom/blob/master/pynetdicom/dul.py#L64.

I have seen this behavior a bit in production, though it is rare.

Reply:
I can pull this out into a separate issue if you'd like

Reply:
Yeah, I really should fix that. It's probably the last holdover from the legacy pynetdicom.